Add file associations support documentation (#5054)
JetBrains/compose-multiplatform: tutorials/Native_distributions_and_local_execution/README.md

}
```

## File associations

File associations can be added with `fileAssociation` function.
These associations can optionally include custom icons.
Similar to [app icon](#app-icon), these custom icons also have OS-specific formats: `.icns` for macOS, `.ico` for Windows and `.png` for Linux.

It is possible to specify both OS-specific and OS-agnostic file associations.
For OS-agnostic you can specify icon files for all three OSes in-place,

> while for OS-specific there is a shorthand with a single icon file parameter.

Reviewer: ```suggestion
while for OS-specific there is a shorthand with a single `iconFile` parameter.
```

```kotlin
compose.desktop {
    application {
        mainClass = "MainKt"

        nativeDistributions {
            targetFormats(TargetFormat.Dmg, TargetFormat.Msi, TargetFormat.Deb)

            packageName = "file-associations-demo"
            packageVersion = "1.0.0"

            macOS {
                fileAssociation(
                    mimeType = "text/kotlin",
                    extension = "kott",
                    description = "Kotlin Source File",
                    iconFile = project.file("Kotlin_icon_big.icns"),
                )
                fileAssociation(
                    mimeType = "text/kotlin",
                    extension = "ko",
                    description = "Kotlin Source File",
                )
            }

>             fileAssociation(

Reviewer: ```suggestion
            // OS-agnostic file association for .kot files, with icons for all three operating systems.
            fileAssociation(
```

                mimeType = "text/kotlin",
                extension = "kot",
                description = "Kotlin Source File",
                macOSIconFile = project.file("Kotlin_icon_big.icns"),
                windowsIconFile = project.file("Kotlin_icon_big.ico"),
                linuxIconFile = project.file("Kotlin_icon_big.png"),
            )

>             fileAssociation(

Reviewer: ```suggestion
            // OS-agnostic file association for .kottt files, without icons.
            fileAssociation(
```
